### PR TITLE
Set webhook defaults update

### PR DESCRIPTION
**Pull request by maurafortino**
-setWebhookDefaults required to take in a pointer - other wise the fields were not being set on the original webhook.
-divided the function into two separate functions one for V1 defaults and one for V2 defaults otherwise we would have had to do a type assertion within setWebhookDefaults on non pointer structs and then returned an empty interface and done yet another type assertion on the return value to set the registration within the Register interface.

### CODE REVIEW

File: transport.go [hunk 1]
```diff
@@ -218,24 +217,20 @@ type webhookValidator struct {
...
-			r.Until = wv.now().Add(time.Duration(r.Duration))
-		}
-		if requestOriginHost != "" {
-			r.Address = requestOriginHost
-		}
-	case *webhook.RegistrationV2:
-		//TODO: do we have any defaults for RegistrationV2 that need to be set?
-		//webhook-schema shows RetryHint, BatchHint, Webhook.SecretHash, and Payload only will have default values
-		//are we setting those values here?
+func (wv webhookValidator) setV1Defaults(r *webhook.RegistrationV1, requestOriginHost string) {
```
Comment (maurafortino) [inline transport.go:220]: separated into two separate functions for two reasons:
1. will have different fields being defaulted
2. if we kept as one function and used the `any` interface we would have had to also return an `any` interface and that would have required type switch statement both before, during, and  after the function call

File: transport.go [hunk 2]
```diff
@@ -94,7 +94,6 @@ func addWebhookRequestDecoder(config transportConfig) kithttp.DecodeRequestFunc
 		var (
 			v1       *webhook.RegistrationV1
 			v2       *webhook.RegistrationV2
-			whreq    addWebhookRequest
```
Comment (maurafortino) [inline transport.go:97]: unit tests were failing. needed a pointer. removed the initiation of this variable and initialize struct at return instead

File: transport.go [hunk 3]
```diff
@@ -104,21 +103,21 @@ func addWebhookRequestDecoder(config transportConfig) kithttp.DecodeRequestFunc
 			return nil, &erraux.Error{Err: err, Message: "failed getting partnerIDs", Code: http.StatusBadRequest}
 		}
 
-		whreq.owner = getOwner(r.Context())
```
Comment (maurafortino) [inline transport.go:107]: now gets added at return statement

**Review by denopink — APPROVED** [review]
great work! lgtm 🍻

**Pull request merged**